packages/blockly/src/components/custom-block-form.tsx

```diff
@@ -305,45 +594,75 @@ export const CustomBlockForm: React.FC<IProps> = ({ blockType, editingBlock, exi
-            <input
-              type="text"
-              placeholder="Label"
-              value={option.label}
-              onChange={(e) => updateOption(index, "label", e.target.value)}
-            />
+
+      {(formData.type === "action") && (
+        <div className={css.customBlockForm_canHaveChildren} data-testid="section-can-have-children">
+          <label htmlFor="can-have-children">
             <input
-              type="text"
-              placeholder="Value"
-              value={option.value}
-              onChange={(e) => updateOption(index, "value", e.target.value)}
-            />
-            <button type="button" onClick={() => removeOption(index)}>Remove</button>
-          </div>
-        ))}
-        <button type="button" className={css.addOptionButton} onClick={addOption}>Add Option</button>
-      </div>
+              checked={formData.canHaveChildren}
+              data-testid="toggle-canHaveChildren"
+              id="can-have-children"
+              type="checkbox"
+              onChange={(e) => setFormData(prev => ({ ...prev, canHaveChildren: e.target.checked }))}
+            /> 
+            Contains Child Blocks
+          </label>
+        </div>
+      )}   
 
-      {formData.type === "creator" && (
+      {(formData.type === "action" && formData.canHaveChildren) && (
+        <div className={css.customBlockForm_childBlocks} data-testid="section-child-blocks">
+          <label htmlFor="child-blocks">Child Blocks</label>
+          <select
+            data-testid="select-childBlocks"
+            id="child-blocks"
+            multiple
+            size={6}
+            value={formData.childBlocks}
+            onChange={handleChildBlocksMultiSelect}
+          >
+            {availableChildOptions.map(o => (
+              <option key={o.id} value={o.id}>{o.name}</option>
+            ))}
+          </select>
+        </div>
+      )}
+
+      {(formData.type === "creator") && (
         <div className={css.customBlockForm_childBlocks}>
-          <label>Child Setter Blocks</label>
-          <select multiple size={3} value={formData.childBlocks} onChange={(e) => {
-            const selectedOptions = Array.from(e.target.selectedOptions).map(opt => opt.value);
-            setFormData(prev => ({ ...prev, childBlocks: selectedOptions }));
-          }}>
-            {existingBlocks?.filter(b => b.type === "setter").map(block => (
-              <option key={block.id} value={block.id}>
-                {block.name}
-              </option>
+          <label htmlFor="child-blocks">Child Blocks</label>
+          <select
+            id="child-blocks"
+            multiple
+            size={6}
+            value={formData.childBlocks}
+            onChange={handleChildBlocksMultiSelect}
+          >
+            {availableChildOptions.map(o => (
+              <option key={o.id} value={o.id}>{o.name}</option>
             ))}
           </select>
         </div>
       )}
```

Comment:
This is a great example of duplicated code that could easily be combined. In this case, you don't even need to make a separate component, I think you could easily just use `formData.type === "creator" || (formData.type === "action" && formData.canHaveChildren)` as the condition to render the same react component. Of course, making it a separate component could be helpful for making the code more readable and manageable.

Reply:
Good idea 👍 Implemented in the latest changes.